src/app/CommandSender.cpp

```diff
@@ -199,5 +292,71 @@ CHIP_ERROR CommandSender::ProcessCommandDataIB(CommandDataIB::Parser & aCommandE
     return err;
 }
 
+CHIP_ERROR CommandSender::PrepareCommand(const CommandPathParams & aCommandPathParams, bool aStartDataStruct)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    CommandDataIB::Builder commandData;
+
+    err = AllocateBuffer();
+    SuccessOrExit(err);
+
+    //
+    // We must not be in the middle of preparing a command, or having prepared or sent one.
+    //
+    VerifyOrExit(mState == CommandState::Idle, err = CHIP_ERROR_INCORRECT_STATE);
+
+    commandData = mInvokeRequestMessage.GetInvokeRequests().CreateCommandData();
+    err         = commandData.GetError();
+    SuccessOrExit(err);
+
+    err = ConstructCommandPath(aCommandPathParams, commandData.CreatePath());
+    SuccessOrExit(err);
+
+    if (aStartDataStruct)
+    {
+        err = commandData.GetWriter()->StartContainer(TLV::ContextTag(to_underlying(CommandDataIB::Tag::kData)),
+                                                      TLV::kTLVType_Structure, mDataElementContainerType);
+    }
+
+    MoveToState(CommandState::AddingCommand);
+
+exit:
```

Comment:
This exit can also be saved if use ReturnErrorOnFailure instead of SuccessOrExit

Reply:
Resolved.